cmd/maya-apiserver/app/server/snapshot_endpoint_v1alpha1.go

```diff
@@ -0,0 +1,213 @@
+	"github.com/openebs/maya/pkg/snapshot/v1alpha1"
+)
+
+type snapshotAPIOps struct {
+	req  *http.Request
+	resp http.ResponseWriter
+}
+
+// SnapshotSpecificRequest deals with snapshot API request w.r.t a Volume
+func (s *HTTPServer) snapshotRequest(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	snapOp := &snapshotAPIOps{
+		req:  req,
+		resp: resp,
+	}
+
+	// The params extracted below are going to be used for RUD operations
+	// volName is the volume name in the query params
+	volName := req.URL.Query().Get("volume")
+	// namespace is the namespace of volume in the query params
+	namespace := req.URL.Query().Get("namespace")
+	// casType is the cas type of volume in the query params
+	casType := req.URL.Query().Get("casType")
+	// snapName is expected to be used only in case of delete and get of a particular snapshot
+	// TODO: Use some http framework to extract snapshot name. strings method is not a good way
+	snapName := strings.Split(strings.TrimPrefix(req.URL.Path, "/latest/snapshots/"), "?")[0]
+
+	switch req.Method {
+	case "POST":
+		return snapOp.create(resp, req)
+	case "GET":
+		// If snapshot name is missing, assume it to be list request
+		if snapName == "" {
+			return snapOp.list(resp, req, volName, namespace, casType)
+		}
+		return snapOp.get(resp, req, snapName, volName, namespace, casType)
+	case "DELETE":
+		return snapOp.delete(resp, req, snapName, volName, namespace, casType)
+	}
+	return nil, CodedError(405, ErrInvalidMethod)
+}
+
+// list is http handler for listing all created snapshot specific to particular volume
+func (sOps *snapshotAPIOps) list(resp http.ResponseWriter, req *http.Request, volName, namespace, casType string) (interface{}, error) {
+	glog.Infof("Snapshot list request was received")
+
+	// Volume name is expected
+	if len(strings.TrimSpace(volName)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("failed to list snapshot: missing snapshot name "))
+	}
+
+	// namespace is expected
+	if len(strings.TrimSpace(namespace)) == 0 {
+		return nil, CodedError(400, fmt.Sprintf("failed to list snapshot: missing namespace "))
+	}
+
+	glog.Infof("Listing snapshots for volume %q ", volName)
+
+	snapOps, err := snapshot.Snapshot(&v1alpha1.SnapshotOptions{
+		CasType:    casType,
+		Namespace:  namespace,
+		VolumeName: volName,
+	})
+	if err != nil {
+		return nil, CodedError(400, err.Error())
+	}
+
+	snaps, err := snapOps.List()
+	if err != nil {
+		glog.Errorf("Failed to list snapshots: error '%s'", err.Error())
+		return nil, CodedError(500, err.Error())
+	}
+
+	glog.Infof("Snapshots listed successfully for volume '%s'", volName)
+	return snaps, nil
+}
+
+// Create is http handler which handles snaphsot-create request
+func (sOps *snapshotAPIOps) create(resp http.ResponseWriter, req *http.Request) (interface{}, error) {
+	glog.Infof("Snapshot create request was received")
+
+	snap := &v1alpha1.CASSnapshot{}
+
+	err := decodeBody(req, snap)
```

Comment:
We are already having `snapshotAPIOps` struct .... so we can use `sOps.req` and `sOps.resp`  further.
but this we can change in next upcoming PR's

Reply:
Will update.